File: dmn-test-cases/standard/tck/1.3/cl3/0004-lending/translator/expected/java/dmn/type/TRequestedProductImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "tRequestedProduct"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class TRequestedProductImpl implements TRequestedProduct {
        private String productType;
        private java.lang.Number amount;
        private java.lang.Number rate;
        private java.lang.Number term;

    public TRequestedProductImpl() {
    }

    public TRequestedProductImpl(java.lang.Number amount, String productType, java.lang.Number rate, java.lang.Number term) {
        this.setAmount(amount);
        this.setProductType(productType);
        this.setRate(rate);
        this.setTerm(term);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Amount")
    public java.lang.Number getAmount() {
        return this.amount;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("Amount")
    public void setAmount(java.lang.Number amount) {
        this.amount = amount;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("ProductType")
    public String getProductType() {
        return this.productType;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("ProductType")
    public void setProductType(String productType) {
        this.productType = productType;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Rate")
    public java.lang.Number getRate() {
        return this.rate;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("Rate")
    public void setRate(java.lang.Number rate) {
        this.rate = rate;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Term")
    public java.lang.Number getTerm() {
        return this.term;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("Term")
    public void setTerm(java.lang.Number term) {
        this.term = term;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-test-cases/standard/tck/1.1/cl3/0004-lending/translator/expected/kotlin/dmn/type/TRequestedProductImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "tRequestedProduct"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class TRequestedProductImpl : TRequestedProduct {
    @get:com.fasterxml.jackson.annotation.JsonGetter("ProductType")
    @set:com.fasterxml.jackson.annotation.JsonGetter("ProductType")
    override var productType: String? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("Amount")
    @set:com.fasterxml.jackson.annotation.JsonGetter("Amount")
    override var amount: kotlin.Number? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("Rate")
    @set:com.fasterxml.jackson.annotation.JsonGetter("Rate")
    override var rate: kotlin.Number? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("Term")
    @set:com.fasterxml.jackson.annotation.JsonGetter("Term")
    override var term: kotlin.Number? = null

    constructor() {
    }

    constructor (amount: kotlin.Number?, productType: String?, rate: kotlin.Number?, term: kotlin.Number?) {
        this.amount = amount
        this.productType = productType
        this.rate = rate
        this.term = term
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}
